Libplanet/Net/Protocols/KademliaProtocol.cs

```diff
@@ -277,6 +277,67 @@ public string Trace()
             return $"Total peer count: {count}\n{trace.Trim('\n')}";
         }
 
+        /// <summary>
+        /// Send <see cref="FindNeighbors"/> messages to <paramref name="viaPeer"/>
+        /// to find <see cref="Peer"/>s near <paramref name="target"/>
+        /// to see if the <see cref="Peer"/>s have a specific address.
+        /// </summary>
+        /// <param name="history">The <see cref="Peer"/> that searched.</param>
+        /// <param name="target">The <see cref="Address"/> to find.</param>
+        /// <param name="viaPeer">The target <see cref="Peer"/> to send <see cref="FindNeighbors"/>
+        /// message. If null, selects 3 <see cref="Peer"/>s from <see cref="RoutingTable"/> of
+        /// self.</param>
+        /// <param name="depth">Target depth of recursive operation.</param>
+        /// <param name="searchTarget">The address to search for among peers.</param>
+        /// <param name="timeout"><see cref="TimeSpan"/> for waiting reply of
+        /// <see cref="FindNeighbors"/>.</param>
+        /// <param name="cancellationToken">A cancellation token used to propagate notification
+        /// that this operation should be canceled.</param>
+        /// <returns>BoundPeer containing the <paramref name="searchTarget"/>'s address.</returns>
+        public async Task<BoundPeer> FindSpecificPeerAsync(
+            ConcurrentBag<BoundPeer> history,
+            Address target,
+            BoundPeer viaPeer,
+            int depth,
+            Address searchTarget,
+            TimeSpan? timeout,
+            CancellationToken cancellationToken)
+        {
+            _logger.Debug(
+                $"{nameof(FindSpecificPeerAsync)}() with {{Target}} to {{Peer}}. " +
+                "(depth: {Depth})",
+                target,
+                viaPeer,
+                depth);
+
+            if (depth == 0)
```

Comment:
Now, if depth is 0, it returns null and exit `FindSpecificPeerAsync()`. So it looks like some breakpoint, right?
If then, IMO it will not work as well. Because `depth` will stay away from 0 by the line (i.e. [KademliaProtocol.cs#L757](https://github.com/planetarium/libplanet/pull/580/files#diff-ae527d5afcdf41da4a57b594cab3cfefR757)).

Reply:
Actually, `depth` became to decrease every step since #665. But this file does not behave as it.